platform/fabric/core/generic/config/service.go

```diff
@@ -0,0 +1,360 @@
+}
+
+type Service struct {
+	name          string
+	prefix        string
+	configService configService
+
+	configuredOrderers []*grpc.ConnectionConfig
+	orderers           []*grpc.ConnectionConfig
+	peerMapping        map[driver.PeerFunctionType][]*grpc.ConnectionConfig
+	channels           []*Channel
+	channelConfigs     []driver.ChannelConfig
+	channelIDs         []string
+	defaultChannel     string
+}
+
+func NewService(configService configService, name string, defaultConfig bool) (*Service, error) {
+	// instantiate base service
+	var service *Service
+	if configService.IsSet("fabric." + name) {
+		service = &Service{
+			name:          name,
+			prefix:        name + ".",
+			configService: configService,
+		}
+	} else {
+		if defaultConfig {
+			service = &Service{
+				name:          name,
+				prefix:        "",
+				configService: configService,
+			}
+		} else {
+			return nil, errors.Errorf("configuration for [%s] not found", name)
+		}
+	}
+	// populate it
+	if err := service.init(); err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+func (s *Service) init() error {
+	// orderers
+	var orderers []*grpc.ConnectionConfig
+	if err := s.configService.UnmarshalKey("fabric."+s.prefix+"orderers", &orderers); err != nil {
+		return err
+	}
+	tlsEnabled := s.TLSEnabled()
+	for _, v := range orderers {
+		v.TLSEnabled = tlsEnabled
+	}
+	s.configuredOrderers = orderers
+	s.orderers = orderers
+
+	// peers
+	var peers []*grpc.ConnectionConfig
+	if err := s.configService.UnmarshalKey("fabric."+s.prefix+"peers", &peers); err != nil {
+		return err
+	}
+
+	peerMapping := map[driver.PeerFunctionType][]*grpc.ConnectionConfig{}
+	for _, v := range peers {
+		v.TLSEnabled = tlsEnabled
+		if v.TLSDisabled {
+			v.TLSEnabled = false
+		}
+		usage := strings.ToLower(v.Usage)
+		switch {
+		case len(usage) == 0:
+			peerMapping[driver.PeerForAnything] = append(peerMapping[driver.PeerForAnything], v)
+		case usage == "delivery":
+			peerMapping[driver.PeerForDelivery] = append(peerMapping[driver.PeerForDelivery], v)
+		case usage == "discovery":
+			peerMapping[driver.PeerForDiscovery] = append(peerMapping[driver.PeerForDiscovery], v)
+		case usage == "finality":
+			peerMapping[driver.PeerForFinality] = append(peerMapping[driver.PeerForFinality], v)
+		case usage == "query":
+			peerMapping[driver.PeerForQuery] = append(peerMapping[driver.PeerForQuery], v)
+		default:
+			logger.Warn("connection usage [%s] not recognized [%v]", usage, v)
+		}
+	}
+	s.peerMapping = peerMapping
+
+	// channels
+	var channels []*Channel
+	var channelIDs []string
+	var channelConfigs []driver.ChannelConfig
+	if err := s.configService.UnmarshalKey("fabric."+s.prefix+"channels", &channels); err != nil {
+		return err
+	}
+	for _, channel := range channels {
+		if err := channel.Verify(); err != nil {
+			return err
+		}
+		channelIDs = append(channelIDs, channel.Name)
+		channelConfigs = append(channelConfigs, channel)
+	}
+	s.channels = channels
```

Comment:
Given how these fields are used I would propose to:
* convert channelConfigs and channels into maps and expose the methods `Channel(name string)` and `ChannelConfig(name string)`.
* remove channelIDs, because it's duplicated information. When you need them you can iterate through the map keys.